src/map.rs

```diff
@@ -937,6 +937,74 @@ where
         n.leading_zeros() as isize | (1 << (RESIZE_STAMP_BITS - 1)) as isize
     }
 
+    // TODO
+    fn try_presize<'g>(&self, size: usize, guard: &'g Guard) {
+        let new_capacity = if size >= MAXIMUM_CAPACITY / 2 {
+            MAXIMUM_CAPACITY
+        } else {
+            // 1.5 * size + 1
+            let size = size + (size >> 1) + 1;
+
+            std::cmp::min(MAXIMUM_CAPACITY, size.next_power_of_two())
+        };
+
+        let c = new_capacity as isize; // TODO
+
+        loop {
+            let sc = self.size_ctl.load(Ordering::SeqCst);
+            if sc < 0 {
+                break;
+            }
+
+            let table = self.table.load(Ordering::SeqCst, &guard);
+
+            // TODO: clean this up?
+            let n = if table.is_null() {
+                None
+            } else {
+                Some(unsafe { table.deref() }.len())
+            };
+
+            if n.is_none() || n.unwrap() == 0 {
+                let n = sc.max(c) as usize;
+
+                let mut table = self.table.load(Ordering::SeqCst, guard);
+
+                if self.size_ctl.compare_and_swap(sc, -1, Ordering::SeqCst) == sc {
+                    if self.table.load(Ordering::SeqCst, guard) == table {
+                        let new_table = Owned::new(Table::new(n));
+                        table = new_table.into_shared(guard);
+                        self.table.store(table, Ordering::SeqCst);
+                        self.size_ctl
```

Comment:
This should change `sc`, not `self.size_ctl` directly. `self.size_ctl` is modified just below the `if`.

Reply:
You're totally correct, I'll fix that now